### PR TITLE
use Make for OSM & GTFS transformations as well, simplify setup

### PR DESCRIPTION
**Pull request by derhuerst**
This PR changes quite a lot of stuff, I'm sorry for that! I tried to keep all commits as atomic as possible.

Some of the benefits of the new structure:

- **less complex, more maintainable**
	- The "old" setup had a shell script that did things, called `make` which did things, which called other shell scripts that did things.
	- Some of the tasks were done twice (e.g. running pfaedle).
	- Some of the responsiblities where split across several places (e.g. patching individual GTFS feeds).
- **more reliable processing** – In multiple places, I have changed the pipeline to abort properly on errors, and to work better in non-GNU/Linux environments.
- support for **running without Docker** – I have written the `makefile` in a way that allows running both inside and outside of Docker: By extracting more processing commands into Make variables, we can easily override them and provide a non-Dockerized version. This is in preparation to get rid of the Docker-within-Docker setup.

The Docker-within-Docker setup is still complex, I couldn't find a way to to make it simpler. Later on, we might create a `gtfs-hub` build image that contains all necessary tools directly, so that we can call them directly.

Please review 6522c23, c08616b, 70c33ae & 98058d9 with special care, as this is where the complex refactoring happens.

I'll go over the commits in their order:

- e57214b, 3adb824, 111eddf, b27fe0c, cc32122 – I tried to update the GTFS URLs to the latest canonical source that I could find (before, they `404`ed); Some providers ceased operations, so I removed them. Although I removed NVV, we should find a way to integrate it; Their feeds on `opendata-oepnv.de` currently fail with `403`.
- e2e9fa5, d366156, 7f1aada – Tweak minor things.
- 70ad84f – Switching from `curl` to `wget` gives us a few useful features out-of-the-box: It can continue downloads, update local files using [HTTP caching](https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching), follow redirects automatically, etc.
- 6522c23 changes the OSM download & processing to use Make targets.
- c08616b changes all of the GTFS downloading & processing (except the `index.html` generation) to use Make targets.
- 70c33ae refactors the `index.html` generation into a single-purpose shell script & a Make target.
- 98058d9 changes the Docker-within-Docker setup – Before, it was AFAICT not impossible to mount a custom config, because `$HOST_DATA` was supposed to point to the host's `data` directory. Now, users have to pass in a value for `$HOST_DIR`, pointing to the GTFS-Hub directory.

---

closes #8

### CODE REVIEW

File: patch_gtfs.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+name=$1
+gtfs_dir=$2
+
+if [ "$1" == "DELFI" ]; then
+  echo "patching DELFI $gtfs_dir/agency.txt"
+  sed -i 's/"","Europe/"https:\/\/www.delfi.de\/","Europe/' "$gtfs_dir/agency.txt"
+fi
+
+if [ "$1" == "VVS" ]; then
+  echo "deleting broken VVS $gtfs_dir/agency.txt"
```
Comment (hbruch) [inline patch_gtfs.sh:15]: The vvs transfers.txt is broken

**Comment by hbruch** [comment]
Executing make all fails with
`make: *** No rule to make target 'data/gtfs/DING.gtfs', needed by 'data/gtfs/ulm.merged.gtfs'.  Stop.`

**Comment by leonardehrenfried** [comment]
When can I expect this to be merged?

**Comment by hbruch** [comment]
Today

**Pull request merged**